binary/expression.proto

```diff
@@ -44,7 +44,8 @@ message Expression {
       string string = 12;
       bytes binary = 13;
       fixed64 timestamp = 14;
-      fixed32 date = 16;
+      // Date in units of milliseconds since the UNIX epoch.
+      int64 date = 16;
```

Comment:
if we use microseconds everywhere else (i believe we do), i think we should use them here as well.

Do you want to take a pass on all of these for comments to make sure we're consistent?

Also, we should rationalize whether we should use fixed64 for this as well (or switch timestamp other to int64)

Reply:
> if we use microseconds everywhere else (i believe we do), i think we should use them here as well.

Makes sense.

> Do you want to take a pass on all of these for comments to make sure we're consistent?

Yup, will do.

> Also, we should rationalize whether we should use fixed64 for this as well (or switch timestamp other to int64)

Will comment on this in a bit.

Reply:
@jacques-n Are these literals allowed to be negative? Currently they aren't by virtue of `fixed32`/`fixed64` being unsigned.

Reply:
Oversight. I guess I wanted `sfixed64`

Reply:
But it may be just a stupid optimization that confuses people. Probably makes sense to just go with int64 even if it will typically be a large number (thus the inclination towards a fixed width value)

Reply:
Ok, I'll revamp the temporal things. Stand by.

Reply:
Revamped.